lib/weather_report/station.ex

```diff
@@ -46,12 +46,26 @@ defmodule WeatherReport.Station do
     |> Stream.map(&station_xmapper/1)
     |> Stream.map(&update_coordinate(&1, :latitude))
     |> Stream.map(&update_coordinate(&1, :longitude))
+    |> Stream.map(&update_feed_urls/1)
     |> Stream.map(&struct(__MODULE__, &1))
     |> Enum.to_list()
   end
 
   defp station_xmapper({:station, doc}), do: SweetXml.xmap(doc, @xmap)
 
+  defp update_feed_urls(station) do
+    station
+    |> Map.update!(:rss_url, &append_subdomain/1)
+    |> Map.update!(:xml_url, &append_subdomain/1)
+  end
+
+  defp append_subdomain(feed_url) do
+    URI.parse(feed_url)
+    |> Map.update!(:host, &("w1." <> &1))
+    |> Map.update!(:authority, &("w1." <> &1))
+    |> URI.to_string()
+  end
+
```

Comment:
Dunno if there's a more idiomatic way to do this. Parsing the url seemed to be safest.

Reply:
It's really odd that the feed of stations is returning data that is just wrong (missing the w1). This looks good to me. :authority is apparently a deprecated field so not technically necessary to update.